### PR TITLE
Feature/65 timeline component 

### PR DESCRIPTION
**Pull request by EeveesEyes**
Implements Timeline from issue #65. It also contains some refactoring.
Please test the component carefully to find probable edge cases /bugs :)

### CODE REVIEW

**Comment by jschurse** [comment]
I searched for the gothic movement:
image

Would it be possible to center the images horizontally in their container?

**Comment by EeveesEyes** [comment]
> Would it be possible to center the images horizontally in their container?

This only happens in Firefox for some reason.

**Comment by jschurse** [comment]
I fixed the horziontal alignment by displaying the image as block and setting the horizontal margin to auto.
@EeveesEyes Could you please test if this is working for you on your devices?
On my Linux it works now with Chrome and Firefox.

File: app/src/app/shared/components/timeline/timeline.component.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    /** Check if event should happen */
+    if (!this.sliderAllowEvent) {
+      this.sliderAllowEvent = true;
+      return;
+    }
+    this.calcSlideStart();
+    this.updateSliderItems();
+
+    /** Set the value to the date of the chosen referenceItem */
+    this.value = +this.items[this.slideStart + this.referenceItem].date;
```
Comment (jschurse) [inline app/src/app/shared/components/timeline/timeline.component.ts]: When running `ng run test` this line throws errors:

```
TypeError: Cannot read property 'date' of undefined

    at TimelineComponent../src/app/shared/components/timeline/timeline.component.ts.TimelineComponent.onSliderMoved (http://localhost:9876/_karma_webpack_/src/app/shared/components/timeline/timeline.component.ts:228:68)

    at Object.eval [as handleEvent] (ng:///DynamicTestModule/TimelineComponent.ngfactory.js:126:31)

    at handleEvent (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/fesm5/core.js:29239:1)

    at callWithDebugContext (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/fesm5/core.js:30309:1)

    at Object.debugHandleEvent [as handleEvent] (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/fesm5/core.js:30036:1)

    at dispatchEvent (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/fesm5/core.js:19859:1)

    at http://localhost:9876/_karma_webpack_/node_modules/@angular/core/fesm5/core.js:21044:30

    at SafeSubscriber.schedulerFn [as _next] (http://localhost:9876/_karma_webpack_/node_modules/@angular/core/fesm5/core.js:23747:35)

    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm5/internal/Subscriber.js:192:1)

    at SafeSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm5/internal/Subscriber.js:130:1)
```

Comment (mauamy) [reply app/src/app/shared/components/timeline/timeline.component.ts]: I just tested running `npm run test` with the current code and could not reproduce this error, which is kind of weird since we did not touch the timeline.component.ts in our last few commits.

Comment (mauamy) [reply app/src/app/shared/components/timeline/timeline.component.ts]: Ok, I have just overseen it. But in my case it seems like the test is repeated a few times and it passes eventually. Still kind of weird, but I'll see what I can do to fix this error.

Comment (mauamy) [reply app/src/app/shared/components/timeline/timeline.component.ts]: Ok, this error occurs because the `items` attribute of the component is not initialized in the test.
So I came up with two possible solutions for this.
The first on is to initialize a few items (with all required fields) in our test's `beforeEach` method:
```typescript
beforeEach(() => {
    fixture = TestBed.createComponent(TimelineComponent);
    component = fixture.componentInstance;
    component.items = [
      {
        id: "Q18578798",
        label: "The Execution of the Four Crowned Martyrs",
        description: "painting by Niccolò di Pietro Gerini",
        type: "artwork",
        absoluteRank: 37,
        relativeRank: 0.9795063116489386,
        date: 1300
      },
      {
        id: "Q549172",
        label: "Flight into Egypt",
        description: "fresco by Giotto",
        type: "artwork",
        absoluteRank: 46,
        relativeRank: 0.9944890206318238,
        date: 1305
      },
      {
        id: "Q979440",
        label: "Annunciation with St. Margaret and St. Ansanus",
        description: "painting by Simone Martini and Lippo Memmi",
        type: "artwork",
        absoluteRank: 36,
        relativeRank: 0.9762525340301427,
        date: 1333
      },
      {
        id: "Q3815314",
        label: "Head of Christ",
        description: "sculpture by Jaume Cascalls",
        abstract: "The Head of Christ is a Jesus head conserved at the National Art Museum of Catalonia.",
        type: "artwork",
        absoluteRank: 21,
        relativeRank: 0.22987683344390347,
        date: 1352,
      },
    ];
    fixture.detectChanges();
  });
```
The second solution would be to only show our component if our items are initialized:
```html
<div *ngIf="items?.length > 0" class="timeline-container">
(...)
</div>
```

I think the best solution would be to combine these two solutions by moving the items initialization to another test case and also apply the second solution. This way we can test if the component is successfully created for both cases (initialized and not initialized items).
But I would like to hear your opinions on that.

Comment (jschurse) [reply app/src/app/shared/components/timeline/timeline.component.ts]: I think to test both cases would be the optimal solution.
Please implement it, if it isn't to much effort.

Comment (mauamy) [reply app/src/app/shared/components/timeline/timeline.component.ts]: I just pushed it. So the tests should run now without any errors.

Comment (jschurse) [reply app/src/app/shared/components/timeline/timeline.component.ts]: Thank you very much!
